### ISSUE TITLE
FlurlResponse.GetStringAsync() throws a NullReferenceException when ResponseMessage.Content is null.

### ISSUE THREAD

**Issue by bombpersons** [tmenier/Flurl#606]
[FlurlResponse.GetStringAsync()](https://github.com/tmenier/Flurl/blob/b5716f3f1e2d5c14b58a98a87fa41d1013ff3a35/src/Flurl.Http/FlurlResponse.cs#L201)

This can easily be reproduced by using HttpTest:

```
using (var httpTest = new HttpTest)
{
  httpTest.RespondWith(status: 500);
  var response = await ("whatever.url").PostAsync();
  var responseString = await response.GetStringAsync();
}
```

I'm assuming this is not intentional? It definitely tripped me up as I was using it via FlurlHttpException.GetResponseStringAsync() which indirectly calls FlurlResponse.GetStringAsync().

**Comment by tmenier** [comment]
Thanks for reporting. I agree it should do a null check, but I wonder how likely it is that this would happen in real-world scenarios. Did it happen in real HTTP call for you or just in a test? It might be that in a response with an empty body normally returns an empty stream for `Content` rather than null, in which case I might want to fix that in `HttpTest` as well. But I'm not sure.

**Comment by bombpersons** [comment]
Only happened in my test. I can't imagine it would happen all that often in the real world. It would definitely trip anyone up if using httptest without specifying content though. Perhaps it should put an empty content by default rather than null?

**Comment by tmenier** [comment]
Agreed. I think I'll do both. HttpTest should probably default it to empty instead of null, but since Content can _technically_ be null (i.e. I can explicitly create an `HttpResponseMessage` that way), I'm good with doing a null check there as well. Thanks for reporting.

**Comment by bombpersons** [comment]
No problem! Thanks for addressing it so quickly =)

**Comment by tmenier** [comment]
Released in 3.2.4. As a side note, I discovered that HttpClient's GetStringAsync shortcut will return an empty string even if the entire HttpContent is null, so I made sure Flurl does the same. Seems the least surprising.
